cpmpy/solvers/choco.py

```diff
@@ -535,6 +536,15 @@ def _get_constraint(self, cpm_expr):
                 # Create task variables. Choco can create them only one by one
                 tasks = [self.chc_model.task(s, d, e) for s, d, e in zip(start, dur, end)]
                 return self.chc_model.cumulative(tasks, demand, cap)
+            elif cpm_expr.name == "subcircuit":
+                # Successor variables
+                succ = self.solver_vars(cpm_expr.args)
+                # Start index must be in subcircuit and thus cannot selfloop
+                if cpm_expr.startIndex is not None:
+                    self += [(cpm_expr.args[cpm_expr.startIndex] != cpm_expr.startIndex)]
```

Comment:
Coming to think of it: this won't work in reified context... You'll always require the start index constraint to be true at the toplevel of the constraint model.

E..g, take the consrtaint `bv -> SubCircuit(x,y,z, start_idx=0)`
Then, valid solutions are:
```
b x y z
1 1 0 2
1 2 1 0
0 0 0 0
0 1 0 0
0 2 0 0
...
```
But the solution `0 0 0 0` for example is not allowed by this formulation. The restriction should be part of the reification (i.e., be in the _constraining_ part somehow)

Reply:
This was I think an issue that you tried to convey earlier in the day. In my previous implementation I had two global constraint classes for SubCircuit and thus a two-stage approach. One that enforces the start index and always decomposes (so that start index restriction can be put in constraining) and a helper global constraint that actually implements the subcircuit (not to be accessed by the user). This way we can decompose for the start index restriction (into the start index constraint and the helper global constraint) but still support the solver's subcircuit. Is this something you're looking for or is there a better way to add to the constraining part afterwards?

Reply:
For now I had not yet listed subcircuit in the list of supported_reified for choco